modules/standard/Sort.chpl

```diff
@@ -145,8 +210,20 @@ implemented with a compare method:
   // This will output: -1, 2, 3, -4
   writeln(Array);
 
+
+The keyPartComparator interface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The keyPartComparator interface defines how a comparator should sort parts of
+a key using the ``keyPart`` method. This is used for certain sort algorithms.
+Records implementing this interface must define a ``keyPart`` method.
+
+A comparator implementing this interface can contain both ``compare`` and
+``keyPart`` methods. In that event, the sort algorithm will use whichever is
+appropriate for the algorithm and expect that they have consistent results.
+
 The .keyPart method
-~~~~~~~~~~~~~~~~~~~
+*******************
 
 A ``keyPart(elt, i)`` method returns *parts* of key value at a time. This
 interface supports radix sorting for variable length data types, such as
```

Comment:
github won't let me annotate the line number, but @mppf had previously pointed out to me that this example was wrong

```chapel
  proc keyPart(elt: 2*int, i: int) {
    if i > 1 then
      return (-1, 0);
    return (keyPartStatus.returned, elt(i));
  }
```

Reply:
Oh good catch, it should be:
```chapel
  proc keyPart(elt: 2*int, i: int) {
    if i > 1 then
      return (keyPartStatus.pre, 0); // second value is not used

    return (keyPartStatus.returned, elt(i));
  }

```

Right?

Reply:
Yes, AFAIK that is right.